types/api/api.go

```diff
@@ -50,9 +58,14 @@ type ListRequest struct{}
 type GetRequest struct {
 	// Name is the name of the secret to fetch.
 	Name string
-	// Version is the version to fetch, or SecretVersionDefault to let
-	// the server pick a version.
+	// Version is the version to fetch, or SecretVersionDefault to let the
+	// server pick a version.
 	Version SecretVersion
+	// Digest is an optional SHA-256 of the current secret value known to the
+	// caller.  If it is set, the server will send a secret value only if the
+	// current value for the requested version has a different digest.  If the
+	// digests match, the server will report an ErrValueNotChanged.
```

Comment:
Server reports an HTTP 304 at this layer, only our Go client would report ErrValueNotChanged.

Reply:
Good call. Fixed, and updated the client to document this error.